cvmfs/util/posix.cc

```diff
@@ -540,8 +545,15 @@ void ReadHalfPipe(int fd, void *buf, size_t nbyte) {
       SafeSleepMs(backoff_ms);
       if (backoff_ms < max_backoff_ms) backoff_ms *= 2;
     }
+  if (timeout_ms != 0) {
+    duration_ms = (platform_monotonic_time_ns() - timestamp) / 1000;
```

Comment:
This is microseconds, not milliseconds

Reply:
Good catch, thanks! I think this also explains the test failures - these were due to the PANIC in ReadHalfPipe being triggered on some reloads.